src/components/CustomerQuota/NoQuota/NoQuotaCard.tsx

```diff
@@ -0,0 +1,210 @@
+    const { transactions, hasLatestTransaction } = value;
+    if (hasLatestTransaction) {
+      latestTransactionsByCategory.push({
+        category: key,
+        transactions: transactions
+      });
+    } else {
+      otherTransactionsByCategory.push({ category: key, transactions });
+    }
+  });
+
+  // Order each category list by alphabetical order
+  const sortTransactionsByCategory = (
+    a: TransactionsByCategory,
+    b: TransactionsByCategory
+  ): 1 | -1 | 0 =>
+    a.category > b.category ? 1 : b.category > a.category ? -1 : 0;
+  latestTransactionsByCategory.sort(sortTransactionsByCategory);
+  otherTransactionsByCategory.sort(sortTransactionsByCategory);
+
+  // Concat category lists into a single list
+  // Add order to each transaction for "Show more" feature
+  let transactionCounter = 0;
+  const transactionsByCategoryList = latestTransactionsByCategory
+    .concat(otherTransactionsByCategory)
+    .map(transactionsByCategory => {
+      const orderedTransactions = transactionsByCategory.transactions.map(
+        transaction => {
+          const result = { ...transaction, order: transactionCounter };
+          transactionCounter += 1;
+          return result;
+        }
+      );
+      return { ...transactionsByCategory, transactions: orderedTransactions };
+    });
+
+  return (
+    <View>
+      <CustomerCard ids={ids} headerBackgroundColor={color("red", 60)}>
+        <View
+          style={[
+            sharedStyles.resultWrapper,
+            sharedStyles.failureResultWrapper
+          ]}
+        >
+          <FontAwesome
+            name="thumbs-down"
+            color={color("red", 60)}
+            style={sharedStyles.icon}
+          />
+          <AppText style={sharedStyles.statusTitleWrapper}>
+            {secondsFromLatestTransaction > 0 ? (
+              secondsFromLatestTransaction > DURATION_THRESHOLD_SECONDS ? (
+                <DistantTransactionTitle
+                  transactionTime={latestTransactionTime!}
+                />
+              ) : (
+                <RecentTransactionTitle
+                  now={now}
+                  transactionTime={latestTransactionTime!}
+                />
+              )
+            ) : (
+              <NoPreviousTransactionTitle />
+            )}
+          </AppText>
+          {transactionsByCategoryList.length > 0 && (
+            <View>
+              <AppText style={styles.wrapper}>
+                Item(s) {policyType === "REDEEM" ? "redeemed" : "purchased"}{" "}
+                previously:
+              </AppText>
+              {transactionsByCategoryList.map(
+                (
+                  transactionsByCategory: TransactionsByCategory,
+                  index: number
+                ) => (
+                  <TransactionsByCategory
+                    key={index}
+                    maxTransactionsToDisplay={
+                      isShowFullList ? 99999 : MAX_TRANSACTIONS_TO_DISPLAY
```

Comment:
We probably need to handle pagination especially for recurring campaigns such as FFTH where the user could be using it for many months, any thoughts on how we should deal with this?

Reply:
Does the policy refresh period last for months? Not sure what is the policy requirement for FFTH. The past transactions endpoint only returns transactions within the current policy period

Reply:
Oh right ok, thought there was still that usecase to show all. Why do we need the `99999` here?

Reply:
Eh, the "show more"? We show first 5 transactions by default, and show everything within policy refresh period when user clicks "Show more"

Reply:
Would it be cleaner if `maxTransactionsToDisplay` is optional, then we can avoid this big number?

Reply:
Hmm I did consider this but that will make for more complex conditional display logic in `TransactionsByCategory`. It would look like

```
export const TransactionsByCategory: FunctionComponent<{
  category: string;
  transactions: Transaction[];
  maxTransactionsToDisplay: number | undefined;
}> = ({ category, transactions, maxTransactionsToDisplay }) => {
  const shouldShowCategory =
    maxTransactionsToDisplay === undefined ||
    (transactions[0].order || 0) < maxTransactionsToDisplay;
  return shouldShowCategory ? (
    <View style={styles.wrapper}>
      <View style={styles.itemRow}>
        <AppText style={styles.itemHeader}>{category}</AppText>
      </View>
      {transactions.map(
        (transaction, index) =>
          (maxTransactionsToDisplay === undefined ||
            (transaction.order || 0) < maxTransactionsToDisplay) && (
            <Transaction key={index} {...transaction} />
          )
      )}
    </View>
  ) : null;
};

```

Seems to me it is quite prone to mistake. Of course we can do testing for it but am not sure if it might be cleaner just to pass in the big number. 

I previously declared this as a constant instead. Does that make it better (since no more magic number in the middle of code)?

Reply:
Yep ok, thanks for explaining the tradeoffs!